File: frontend/src/types/template.ts
import type { PlanType } from "./plan";

export interface SectionTemplate {
  key: string;
  title: string;
  level: number;
  sort_order: number;
  ai_generatable: boolean;
  user_editable: boolean;
  required: boolean;
  auto_fill: boolean;
  auto_fill_source: string | null;
  gb_requirement: string;
  prompt_template: string | null;
  data_dependencies: string[];
  subsections: SectionTemplate[];
}

export interface PlanTemplate {
  id: string;
  plan_type: PlanType;
  name: string;
  version: string;
  structure: SectionTemplate[];
  is_active: boolean;
}

File: backend/app/schemas/template.py
from pydantic import BaseModel

class SectionTemplateItem(BaseModel):
    key: str; title: str; level: int; sort_order: int
    ai_generatable: bool = True; user_editable: bool = True; required: bool = True
    auto_fill: bool = False; auto_fill_source: str | None = None
    gb_requirement: str = ""; prompt_template: str | None = None
    data_dependencies: list[str] = []; subsections: list["SectionTemplateItem"] = []

class TemplateResponse(BaseModel):
    id: str; plan_type: str; name: str; version: str; structure: list = []; is_active: bool
    model_config = {"from_attributes": True}
